datafusion/expr/src/utils.rs

```diff
@@ -69,6 +69,94 @@ pub fn grouping_set_expr_count(group_expr: &[Expr]) -> Result<usize> {
     }
 }
 
+fn powerset<T>(slice: &[T]) -> Vec<Vec<&T>> {
+    let mut v = Vec::new();
+    for mask in 0..(1 << slice.len()) {
```

Comment:
I think that this should verify that that slice.len() is less than 64 and return an error  to avoid overflow / panic?

Reply:
You are so right, thanks for the reminder，i will check it